### PR TITLE
Add APM documentation

### PR DESCRIPTION
**Pull request by dmikusa**
This PR makes two changes:

1. It adds a dedicated section for documentation around the APM tools buildpacks. This provides instructions for each supported APM integration, as well as two general examples for using the APM tools buildpacks.
2. It removes the section under the Java docs, which previously had APM instructions, and references the newly created APM docs section.

Signed-off-by: Daniel Mikusa <dmikusa@vmware.com>

### CODE REVIEW

File: content/docs/howto/app-monitor.md
Comment (pivotal-david-osullivan) [inline content/docs/howto/app-monitor.md:54]:
```suggestion
The Datadog agent requires a side-car agent to be running in addition to the Java agent. This side-car agent runs outside of the buildpack generated image. The [standard Datadog instructions for your container orchestrator of choice](https://docs.datadoghq.com/tracing/setup_overview/setup/java/?tab=containers#configure-the-datadog-agent-for-apm) can be used to install this agent. The Paketo team also has detailed instructions for [various runtimes available here](https://github.com/paketo-buildpacks/datadog/blob/main/docs/run-on-docker.md).
```
Should this link to here instead? It says various runtimes but there's only Docker instructions there

File: content/docs/howto/app-monitor.md
Comment (pivotal-david-osullivan) [inline content/docs/howto/app-monitor.md:191]:
```suggestion
  --volume "$(pwd)/binding:/bindings/application-insights" \
```

File: content/docs/howto/app-monitor.md
Comment (pivotal-david-osullivan) [inline content/docs/howto/app-monitor.md:188]: This doesn't seem to work, according to docs it needs to be given a 'Connection String' which is then set to a value of InstrumentationKey=<blah>
```suggestion
echo "InstrumentationKey=<key>" > binding/CONNECTION_STRING
```
It must be capitalised too

File: content/docs/howto/app-monitor.md
Comment (pivotal-david-osullivan) [inline content/docs/howto/app-monitor.md:190]: Not a big deal if you follow it properly but can this be shown without overriding the SERVICE_BINDING_ROOT? I feel like bindings are tricky as it is

**Review by pivotal-david-osullivan — CHANGES_REQUESTED** [review]

File: content/docs/howto/app-monitor.md
Comment (dmikusa) [reply content/docs/howto/app-monitor.md:54]: That's future-proofing :)  Hopefuly, we'll have instructions for at least Kubernetes as well.

File: content/docs/howto/app-monitor.md
Comment (dmikusa) [reply content/docs/howto/app-monitor.md:188]: OK. This must have changed. I just copied over the previous example. Good catch, thanks.

File: content/docs/howto/app-monitor.md
Comment (dmikusa) [reply content/docs/howto/app-monitor.md:190]: Yes, if you just use `/platform/bindings` it should work w/out that env variable.

File: content/docs/howto/app-monitor.md
Comment (dmikusa) [inline content/docs/howto/app-monitor.md:253]:
```suggestion
[azure application insights connection string]:https://docs.microsoft.com/en-us/azure/azure-monitor/app/sdk-connection-string?tabs=net#finding-my-connection-string
```

File: content/docs/howto/app-monitor.md
Comment (dmikusa) [inline content/docs/howto/app-monitor.md:185]:
```suggestion
3. To connect to Azure Application Insights at runtime a valid [connection string][azure application insights connection string] is required.
```

**Comment by ForestEckhardt** [comment]
Is there anyway we could get the @paketo-buildpacks/app-monitoring-maintainers team to exist before merging this PR?

**Comment by joshuatcasey** [comment]
I hope to review on Tuesday, but please do not block merging on that! I can PR in changes after it's live.

**Comment by dmikusa** [comment]
@ForestEckhardt I did create the App Monitoring teams. Added as a reviewer.

**Comment by joshuatcasey** [comment]
Do we have guidance on how to include these in the non-Java languages? I don't believe any of these buildpacks are included in non-Java language families.

Related: should we seek to include the relevant buildpacks into those language families that they support?

**Comment by dmikusa** [comment]
These have not been added into the Java composite buildpacks because of the layer limits and how close we are to those with the builders.

I would not suggest adding them to other composite Buildpacks for the same reason.

You can use them by simply appending the desired buildpack onto the end of the buildpack list with `pack build -b`.

File: content/docs/howto/app-monitor.md
Comment (fg-j) [inline content/docs/howto/app-monitor.md:8]:
```suggestion
    name: "Enable Application Monitoring"
```
This keeps the verb-phrase style of our other How To menu items.

**Review by fg-j — APPROVED** [review]

**Pull request merged**
